java/emitter.js

```diff
@@ -1,425 +1,484 @@
-      throw Error('instanceOf not implemented in Java')
-    },
-    throws(func, check){
-      let command
-      if(check){
-        command = java`Assert.assertThrows(${insert(check())} , new Assert.ThrowingRunnable(){
+    }
+
+    const eyes = {
+        constructor: {
+            setViewportSize(viewportSize) {
+                return addCommand(java`Eyes.setViewportSize(getDriver(), ${addType(viewportSize, 'RectangleSize')});`)
+            }
+        },
+        runner: {
+            getAllTestResults(throwEx) {
+                return addCommand(java`getRunner().getAllTestResults(${throwEx});`).type('TestResultsSummary').methods({
+                    getAllResults: (target) => addCommand(java`${target}.getAllResults();`).type({
+                        type: 'Array',
+                        items: {
+                            type: 'TestResultContainer'
+                        }
+                    })
+                })
+            }
+        },
+        open({appName, testName, viewportSize}) {
+            let command = []
+            command.push('open(driver')
+            command.push(java`, ${appName || test.config.appName}`)
+            command.push(java`, ${testName || test.config.baselineName}`)
+            if (viewportSize) command.push(java`, new RectangleSize(${viewportSize.width}, ${viewportSize.height})`)
+            command.push(');')
+            addCommand(command.join(''))
+        },
+        check(checkSettings = {}) {
+            if (test.api === 'classic') {
+                if (checkSettings.frames === undefined && checkSettings.region === undefined) {
+                    eyes.checkWindow(checkSettings.tag, checkSettings.matchTimeout, checkSettings.isFully)
+                } else if (checkSettings.frames && checkSettings.region) {
+                    eyes.checkRegionInFrame(checkSettings.frames, checkSettings.region, checkSettings.matchTimeout, checkSettings.tag, checkSettings.isFully)
+                } else if (checkSettings.frames) {
+                    eyes.checkFrame(checkSettings.frames, checkSettings.matchTimeout, checkSettings.tag)
+                } else if (checkSettings.region) {
+                    eyes.checkRegion(checkSettings.region, checkSettings.matchTimeout, checkSettings.tag)
+                } else {
+                    throw new Error('Not implemented classic api method was tried to generate')
+                }
+            } else {
+                addCommand(`eyes.check(${checkSettingsParser(checkSettings, test.meta.native)});`)
+            }
+        },
+        checkWindow(tag, matchTimeout, stitchContent) {
+            if (matchTimeout && stitchContent) throw new Error(`There is no signature in java SDK for usage both matchTimeout and stitchContent`)
+            const commands = []
+            commands.push(java`eyes.checkWindow(`)
+            if (matchTimeout) commands.push(java`${matchTimeout}, `)
+            commands.push(java`${tag || ''}`)
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/sdk.coverage.tests/pullRequest?prid=8528488)
```suggestion
            commands.push(java`${tag || ""}`)
```